pkg/controller/vitesscell/vitesscell_controller.go

```diff
@@ -195,8 +196,28 @@ func (r *ReconcileVitessCell) Reconcile(cctx context.Context, request reconcile.
 		resultBuilder.Error(err)
 	}
 
+	// List all VitessShard in the same cluster, we do this to determine what is the image used for the mysqld container
+	// Note that this is cheap because it comes from the local cache.
+	labels := map[string]string{
+		planetscalev2.ClusterLabel: vtc.Labels[planetscalev2.ClusterLabel],
+	}
+	opts := &client.ListOptions{
+		Namespace:     vtc.Namespace,
+		LabelSelector: apilabels.SelectorFromSet(labels),
+	}
+	vts := &planetscalev2.VitessShardList{}
+	if err := r.client.List(ctx, vts, opts); err != nil {
+		r.recorder.Eventf(vtc, corev1.EventTypeWarning, "ListFailed", "failed to list VitessShard objects: %v", err)
+		return resultBuilder.Error(err)
+	}
+
+	var mysqldImage string
+	if len(vts.Items) > 0 {
+		mysqldImage = vts.Items[0].Spec.Images.Mysqld.Image()
+	}
+
```

Comment:
Leaving `mysqldImage` empty if no shard was found is okay, it will just result in using the default mysql server version